cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2963,6 +2978,87 @@ String getInternalServiceHostname(String serviceName)    {
             return withVoid(Future.succeededFuture());
         }
 
+        Future<ReconciliationState> getJmxTransDescription() {
+            Promise<ReconciliationState> promise = Promise.promise();
+
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
+                future -> {
+                    try {
+                        this.jmxTrans = JmxTrans.fromCrd(kafkaAssembly, versions);
+                        if (this.jmxTrans != null) {
+                            int numOfBrokers = kafkaCluster.getReplicas();
+                            this.jmxTransConfigMap = jmxTrans.generateJmxTransConfigMap(kafkaAssembly.getSpec().getKafka().getJmxOptions().getJmxTransSpec(), numOfBrokers);
+                            this.jmxTransDeployment = jmxTrans.generateDeployment(imagePullPolicy, imagePullSecrets);
+                        }
+                        future.complete(this);
+                    } catch (Throwable e) {
+                        future.fail(e);
+                    }
+                }, true,
+                res -> {
+                    if (res.succeeded()) {
+                        promise.complete(res.result());
+                    } else {
+                        promise.fail(res.cause());
+                    }
+                }
+            );
+            return promise.future();
+        }
+
+        Future<ReconciliationState> jmxTransConfigMap() {
+            return withVoid(configMapOperations.reconcile(namespace,
+                    JmxTrans.jmxTransConfigName(name),
+                    jmxTransConfigMap));
+        }
+
+
+        Future<ReconciliationState> jmxTransServiceAccount() {
+            return withVoid(serviceAccountOperations.reconcile(namespace,
+                    JmxTrans.containerServiceAccountName(name),
+                    jmxTrans != null ? jmxTrans.generateServiceAccount() : null));
+        }
+
+        Future<ReconciliationState> jmxTransDeployment() {
+            if (this.jmxTrans != null && this.jmxTransDeployment != null) {
+                Future<Deployment> deploymentFuture = deploymentOperations.getAsync(namespace, this.jmxTrans.getName());
+                return deploymentFuture.compose(dep -> {
+                    Future<ConfigMap> configMapFuture = configMapOperations.getAsync(namespace, jmxTransConfigMap.getMetadata().getName());
+                    return configMapFuture.compose(res -> {
+                        if (configMapFuture.succeeded()) {
```

Comment:
When you use `compose` ... it should be already succeeded and the `res` should be the result. So IMHO you do not need to check the result and youdo not need to do `configMapFuture.result().getMetadata().getResourceVersion()` but just `res.getMetadata().getResourceVersion()`.

Reply:
I did not know that! Thanks for pointing that out.